Core/src/org/sleuthkit/autopsy/discovery/search/DiscoveryAttributes.java

```diff
@@ -258,37 +258,33 @@ private void processResultFilesForCR(List<Result> results,
             Set<String> hashesToLookUp = new HashSet<>();
             List<ResultDomain> domainsToQuery = new ArrayList<>();
             for (Result result : results) {
-                if (result.getKnown() == TskData.FileKnown.KNOWN) {
-                    result.setFrequency(SearchData.Frequency.KNOWN);
-                }
-
-                if (result.getType() != SearchData.Type.DOMAIN) {
-                    ResultFile file = (ResultFile) result;
-                    if (file.getFrequency() == SearchData.Frequency.UNKNOWN
-                            && file.getFirstInstance().getMd5Hash() != null
-                            && !file.getFirstInstance().getMd5Hash().isEmpty()) {
-                        hashesToLookUp.add(file.getFirstInstance().getMd5Hash());
-                        currentFiles.add(file);
+                // If frequency was already calculated, skip...
+                if (result.getFrequency() == SearchData.Frequency.UNKNOWN) {
+                    if (result.getKnown() == TskData.FileKnown.KNOWN) {
+                        result.setFrequency(SearchData.Frequency.KNOWN);
                     }
 
-                    if (hashesToLookUp.size() >= BATCH_SIZE) {
-                        computeFrequency(hashesToLookUp, currentFiles, centralRepoDb);
+                    if (result.getType() != SearchData.Type.DOMAIN) {
+                        ResultFile file = (ResultFile) result;
+                        if (file.getFirstInstance().getMd5Hash() != null
+                                && !file.getFirstInstance().getMd5Hash().isEmpty()) {
+                            hashesToLookUp.add(file.getFirstInstance().getMd5Hash());
+                            currentFiles.add(file);
+                        }
 
-                        hashesToLookUp.clear();
-                        currentFiles.clear();
-                    }
-                } else {
-                    ResultDomain domainInstance = (ResultDomain) result;
-                    if (domainInstance.getFrequency() != SearchData.Frequency.UNKNOWN) {
```

Comment:
This is important for not re-calculating frequencies. I think it should stay.

Reply:
This check was moved earlier to line 262 as a conditional just inside the for the loop wrapping everything so that frequencies won't be recalculated for files either.

Reply:
I missed that, thanks